### DISCUSSION TITLE
req: Let's Encrypt DNS-01 challenge support

### DISCUSSION THREAD

**Discussion by eelkevdbos** [basecamp/kamal-proxy#44] [Ideas and Issue Triage]
This challenge allows applications behind a firewall that restrict access based on IP-address ranges to still have the means to validate without knowing Let's Encrypt's IP ranges (they do not provide any range of their infra by design).

Since this project is a 'minimal' proxy, I could see that this kind of feature would not be in scope, but I am wondering if there are any plans to extend to features like this - or have a pluggable infrastructure that would allow the community to build something like this.

**Comment by erka** [comment]
I believe this is a part of #30

**Reply by alhafoudh** [reply]
Well, it looks like it is not. :(

**Reply by aareman** [reply]
Yeah, DNS challenge wasn't part of #30

**Comment by kevinmcconnell** [comment]
I'd be interested in ideas for how something pluggable might look. I don't think we'd want to add anything too complex here, but if we had an easy way to call out to a script for the DNS operations, it could be nice to allow people to wire up the details themselves. Particularly because DNS-01 support would mean we could support wildcard certs.

So, definitely seems worth thinking about.
